### DISCUSSION TITLE
Random Error 500 issues

### DISCUSSION THREAD

**Discussion by rastacalavera** [Kovah/LinkAce#599] [Support] [answered]
Hello, I am having issues where as I click around the application, I will randomly get error 500 messages. Refreshing 1-3 times gets me back into the app though. This started happening after I had to bring my host down for an update and then when I brought my docker stack back up, this issue started occurring.

I am using image: linkace/linkace:simple and image: mariadb:10.7 I just updated the docker Linkace image to the one built on december 16th cause I thought maybe I needed to do that? It hasn't helped.

I included a log from the web GUI below

```


Level | Context | Date | Content
-- | -- | -- | --
error | production | 2023-01-18 15:17:44 | SQLSTATE[HY000] [1045] Access denied for  user 'linkace'@'172.31.0.10' (using password: YES) (SQL: select * from  `users` where `id` = 1 limit 1) {"exception":"[object]  (Illuminate\\Database\\QueryException(code: 1045): SQLSTATE[HY000]  [1045] Access denied for user 'linkace'@'172.31.0.10' (using password:  YES) (SQL: select * from `users` where `id` = 1 limit 1) at  /app/vendor/laravel/framework/src/Illuminate/Database/Connection.php:760)
```
my working folder and permissions are shown below:
```
tiny@tiny:~/linkace$ ls -la
total 20
drwxrwxr-x  3 tiny tiny   4096 Jan 18 09:12 .
drwxr-x--- 24 tiny tiny   4096 Jan 17 15:19 ..
drwxrwxr-x  6 lxd  docker 4096 Jan 18 09:12 db
-rwxrwxr-x  1 tiny tiny   1254 Jan 17 15:07 docker-compose.yml
-rwxrwxr-x  1 tiny tiny    823 Jan 18 09:01 .env
```
The PUID and GID in the environment section of the docker compose are 1000 which match my user "tiny", do I maybe need to just change the group and user of the db to match everything else?

**Comment by Kovah** [comment, marked as answer]
> The PUID and GID in the environment section of the docker compose are 1000 which match my user "tiny", do I maybe need to just change the group and user of the db to match everything else?

It could be possible that this is causing issues. If you can use the app normally but occasional those permission errors occur, I assume this is not an issue with Linkace but the database itself.

**Reply by rastacalavera** [reply]
I think I may have changed the appkey along the way in my meddling. I started over from the the beginning with a fresh install and it seems to be holding steady. 

Is there any reason why in the simple docker install, the db volumes are put in the /var/lib/docker/volumes rather than letting the user define where they want the volume to live?

**Reply by Kovah** [reply]
I have chosen to use a volume because it's exactly what a simple Docker setup should look like. Volumes are easy and are usually not deleted automatically. If you need to use a local folder you are free to do that.

**Reply by sliine** [reply]
> I have chosen to use a volume because it's exactly what a simple Docker setup should look like. Volumes are easy and are usually not deleted automatically. If you need to use a local folder you are free to do that.

You can help me? - I created discussions.
